### DISCUSSION TITLE
Default Shutter Speed?

### DISCUSSION THREAD

**Discussion by coult099** [mitsuba-renderer/mitsuba3#1416] [Q&A] [answered]
In the documentation for the `hdrfilm` plugin, I see: "...the output file will record linear radiance values." I take this to mean that each pixel in the rendered image has units of radiance (W/sr/m^2). However, in the sensor source code, there is reference to variables `m_shutter_open` and `m_shutter_open_time`. This leads me to believe that the rendered image actually has units of J/sr/m^2, since we have a time over which the image is captured. Is this correct? Apologies if I have made a basic error here. 

Additionally, I cannot see where the variables `m_shutter_open` and `m_shutter_open_time` are set to a default value. There is code to be able to set these parameters and read them off from a given sensor, but I do not see where or if they have a default value. When calling `mi.render`, is there a default shutter speed? Or, should I scale the image by my desired shutter speed to achieve a given exposure? Any insight is greatly appreciated.

**Comment by rtabbara** [comment, marked as answer]
Hi @coult099 ,

Sensors can be provided with `shutter_open` and `shutter_close` parameters during initialisation. They are both default initialised to `0` in `sensor.cpp`.

However, please keep in mind while Mitsuba 3 does indeed sample rays over the shutter time interval (if you choose to set them), there are currently no native plugins that utilise this information. So for instance, you won't be able to simulate motion blur effects from shapes that move over that time interval.

More importantly, the films that are provided don't simulate the effect of radiant exposure either, so that however long you set your exposure time doesn't affect the final output. You would need to modify or add new sensors to model these additional effects (artifacts) of physical cameras.

**Reply by coult099** [reply]
Thank you for the information. I am not interested in motion blur, but I would like to know how the rendered image differs from an image captured by an ordinary RGB digital camera if the integration/shutter time is not taken into account: should I multiply the rendered image by a desired integration time/shutter open time to simulate the effect of higher or lower exposure?

My specific application area involves observing the effect of Poisson noise on an image, for which I am using the render as ground truth. However, the amount of Poisson noise depends on the integration time/shutter open time of the image, so it is necessary to know this parameter to add a realistic amount of noise to the rendered image. Since Mitsuba 3 does not use shutter time, is there another way to determine how much Poisson noise can be expected in a rendered image? Thank you.

**Reply by njroussel** [reply]
Hi @coult099 

The default values are filled in [`Sensor(const Properties& props)`](https://github.com/mitsuba-renderer/mitsuba3/blob/master/src/render/sensor.cpp#L15-L16).

Mitusba does not model Poisson/shot noise, as you noted, you would need to add this yourself. You can access the exposure time with [`sensor.shutter_open_time()`](https://mitsuba.readthedocs.io/en/stable/src/api_reference.html#mitsuba.Sensor.shutter_open_time).

**Reply by njroussel** [reply]
I think to fully model the effect correctly you'd need to add some code during the film splatting too. In order to track the number of photons deposited, for example.

**Reply by coult099** [reply]
Hi @njroussel,
Thank you for the information. My original thought was to just add the Poisson noise after the render is complete, as a post-processing step. If I knew or could set the exposure time, it should be simple to calculate the Poisson rate parameter based on exposure time. However, since the shutter time defaults to 0, I am not sure how to set the rate parameter for Poisson noise.

Another option is to interpret the pixel values as the means of Poisson noise distributions (with appropriate scaling), and then generate a the noisy image where the pixels are drawn from these distributions. In this case, most of the literature I have seen talks about scaling the mean of the Poisson distribution with the "intensity" of the pixel. As I understand the documentation for `hdrfilm`, however, it seems Mitsuba outputs radiance, not intensity values. Is this a language/terminology issue? Or, is there a way to convert (possibly using some notion of exposure time) the Mitsuba output to intensity?
